bedrock/security/models.py

```diff
@@ -132,3 +132,89 @@ def year_quarter(self):
     def quarter_string(self):
         year, quarter = self.year_quarter
         return '%s Quarter %s' % (self.ORDINALS[quarter], year)
+
+
+class MitreCVE(models.Model):
+    id = models.CharField(max_length=15, primary_key=True, db_index=True)
+    year = models.SmallIntegerField()
+    order = models.SmallIntegerField()
+    title = models.CharField(max_length=200, blank=True)
+    impact = models.CharField(max_length=100, blank=True)
+    reporter = models.CharField(max_length=100, blank=True)
+    description = models.TextField()
+    products = JSONField(default='[]')
+    bugs = JSONField(default='[]')
+
+    class Meta:
+        ordering = ('-year', '-order')
+
+    def product_versions(self):
+        """Return a list of version numbers per product"""
+        prod_vers = {}
+        for prod in self.products:
+            prod_name, version = prod.rsplit(None, 1)
+            if prod_name in prod_vers:
+                prod_vers[prod_name].append(version)
+            else:
+                prod_vers[prod_name] = [version]
+
+        return prod_vers
+
+    def feed_entry(self):
+        """Return a MITRE format data structure for the CVE
+
+        See https://github.com/CVEProject/automation-working-group/blob/master/cve_json_schema/DRAFT-JSON-file-format-v4.md
+        """
+        product_data = []
+        for prod_name, versions in self.product_versions().iteritems():
+            product_data.append({
+                'product_name': prod_name,
+                'version': {
+                    'version_data': [{'version_value': vers} for vers in versions],
+                }
+            })
+
+        return {
+            'data_type': 'CVE',
+            'data_format': 'MITRE',
+            'data_version': '4.0',
+            'CVE_data_meta': {
+                'ID': self.id,
+                'ASSIGNER': 'cve@mitre.org'
+            },
+            'affects': {
+                'vendor': {
+                    'vendor_data': [
+                        {
+                            'vendor_name': 'Mozilla',
+                            'product': {
+                                'product_data': product_data,
+                            }
+                        }
+                    ]
+                }
+            },
+            'problemtype': {
+                'problemtype_data': [
+                    {
+                        'description': [
+                            {
+                                'lang': 'eng',
+                                'value': self.title,
```

Comment:
Problemtype is not the title, they are wanting something more generic like a CWE "Use after free", "buffer overflow", etc. see https://cwe.mitre.org/data/definitions/659.html for a partial list.

Our current yaml doesn't have a clean existing source of this. Some of the right phrases exist in the titles, but sometimes not.

I'm confused by the spec, which says "Mandatory: None", but also that it's a CNA requirement (and it shows up in the minimal examples). I'm tempted to drop it and see what complaints we get, but I suppose title is as good as anything if we don't drop it.

Reply:
We could come up with a yaml entry for `cve_problem_type` or the like. And I can use that if defined or `title` if not. Would that be sufficient?

Reply:
That sounds good. And if the CVE people complain we could easily change it to be cve_problem_type if present and nothing otherwise.

Reply:
Works for me. I'll get this updated and deployed to the demo early next week for you to check out, then hopefully we can merge this shortly thereafter.